onnxruntime/python/tools/symbolic_shape_infer.py

```diff
@@ -877,7 +877,9 @@ def _infer_Expand(self, node):
             # new_shape's dim can come from shape value
             self._update_computed_dims(expand_to_shape)
             shape = self._get_shape(node, 0)
-            new_shape = self._broadcast_shapes(shape, get_shape_from_sympy_shape(expand_to_shape))
+            # we should not update Expand's input's symbolic dims according to the inferred output dims.
+            # because there is a broadcast rule applied on the two inputs.
+            new_shape = self._broadcast_shapes(shape, get_shape_from_sympy_shape(expand_to_shape), add_suggested_merge=False)
```

Comment:
I think this will get this model to pass, but may cause some other models to fail. auto_merge is not guaranteed to be always correct exactly because of symbolic dims could be 1.
As to this model, IMO the fix should be on the model side to avoid symbolic dimensions from input when those dims are fixed to 1.

Reply:
Thanks @KeDengMS for the clarification. This " auto_merge is not guaranteed to be always correct exactly because symbolic dims could be 1 as in this comment" sounds scary :) 

BTW, "may cause some other models to fail", what kinds of issues shall we get? missing shape inferred or wrong shape inferred?

Reply:
Incorrect shape inference could lead to failure in this script, or runtime exception when input mismatches. I remember saw cases of output dim from a Loop op needs to be merged with fixed dims. This may affect TensorRT EP when running some ops.
As to the issue you found, I think maybe better to add a flag for the new behavior, and turn it on for training. This might minimize the impact to other symbolic shape inference scenarios.

Reply:
Got it, thanks Ke. :)